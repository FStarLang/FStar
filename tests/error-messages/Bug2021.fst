module Bug2021

[@(expect_failure [66])]
let test0 () =
  let x = (fun r -> 2) in
  ()

[@(expect_failure [66])]
let test1 () =
  let x r = 2 in
  ()

[@(expect_failure [66])]
let test2 () =
  ();
  let blah #r : int =
    2
  in
  ()

(* Reports over `f` *)
[@(expect_failure [66])]
let test3 (f : (#x:int -> unit -> unit -> unit)) =
  let blah = f () () in
  ()

(* Reports over `f ()` *)
[@(expect_failure [66])]
let test4 (f : (unit -> #x:int -> unit -> unit)) =
  let blah = f () () in
  ()

(* Reports over `f`, since it raises that error first
 * and then typechecking is aborted. *)
[@(expect_failure [66])]
let test5 (f : (#i:int -> unit -> #x:int -> unit -> unit)) =
  let blah = f () () in
  ()
