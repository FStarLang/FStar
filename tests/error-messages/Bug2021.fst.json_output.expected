>> Got issues: [
{"msg":["Failed to resolve implicit argument ?2\nof type Type\nintroduced for user-provided implicit term at Bug2021.fst(5,15-5,16)"],"level":"Error","range":{"def":{"file_name":"Bug2021.fst","start_pos":{"line":5,"col":15},"end_pos":{"line":5,"col":16}},"use":{"file_name":"Bug2021.fst","start_pos":{"line":5,"col":15},"end_pos":{"line":5,"col":16}}},"number":66,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let test0`","While typechecking the top-level declaration `[@@expect_failure] let test0`"]}
>>]
>> Got issues: [
{"msg":["Failed to resolve implicit argument ?2\nof type Type\nintroduced for user-provided implicit term at Bug2021.fst(10,8-10,9)"],"level":"Error","range":{"def":{"file_name":"Bug2021.fst","start_pos":{"line":10,"col":8},"end_pos":{"line":10,"col":9}},"use":{"file_name":"Bug2021.fst","start_pos":{"line":10,"col":8},"end_pos":{"line":10,"col":9}}},"number":66,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let test1`","While typechecking the top-level declaration `[@@expect_failure] let test1`"]}
>>]
>> Got issues: [
{"msg":["Failed to resolve implicit argument ?5\nof type Type\nintroduced for user-provided implicit term at Bug2021.fst(16,12-16,13)"],"level":"Error","range":{"def":{"file_name":"Bug2021.fst","start_pos":{"line":16,"col":12},"end_pos":{"line":16,"col":13}},"use":{"file_name":"Bug2021.fst","start_pos":{"line":16,"col":12},"end_pos":{"line":16,"col":13}}},"number":66,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let test2`","While typechecking the top-level declaration `[@@expect_failure] let test2`"]}
>>]
>> Got issues: [
{"msg":["Failed to resolve implicit argument ?9\nof type Prims.int\nintroduced for Instantiating implicit argument 'x'"],"level":"Error","range":{"def":{"file_name":"Bug2021.fst","start_pos":{"line":23,"col":11},"end_pos":{"line":23,"col":12}},"use":{"file_name":"Bug2021.fst","start_pos":{"line":24,"col":13},"end_pos":{"line":24,"col":14}}},"number":66,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let test3`","While typechecking the top-level declaration `[@@expect_failure] let test3`"]}
>>]
>> Got issues: [
{"msg":["Failed to resolve implicit argument ?10\nof type Prims.int\nintroduced for Instantiating implicit argument 'x'"],"level":"Error","range":{"def":{"file_name":"Bug2021.fst","start_pos":{"line":29,"col":11},"end_pos":{"line":30,"col":17}},"use":{"file_name":"Bug2021.fst","start_pos":{"line":30,"col":13},"end_pos":{"line":30,"col":17}}},"number":66,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let test4`","While typechecking the top-level declaration `[@@expect_failure] let test4`"]}
>>]
>> Got issues: [
{"msg":["Failed to resolve implicit argument ?13\nof type Prims.int\nintroduced for Instantiating implicit argument 'x'"],"level":"Error","range":{"def":{"file_name":"Bug2021.fst","start_pos":{"line":35,"col":11},"end_pos":{"line":36,"col":17}},"use":{"file_name":"Bug2021.fst","start_pos":{"line":36,"col":13},"end_pos":{"line":36,"col":17}}},"number":66,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let test5`","While typechecking the top-level declaration `[@@expect_failure] let test5`"]}
>>]
